app/controllers/api/v1/admin/salaries_controller.rb

```diff
@@ -0,0 +1,50 @@
+class Api::V1::Admin::SalariesController < Api::V1::BaseController
+  before_action :authenticate_user!
+  before_action :load_salary
+
+  def index
+    respond_with @salary.last
+  end
+
+  def create
+    if archive_salary
```

Comment:
I'd recommend adding service object for that (ex: `SalaryCreator`), where we can handle this. Also, it is better to do these changes in the transaction. In the code you have atm there is a possibility when you've successfully archived the old salary and the new salary was not created. After adding `SalaryCreator` service don't forget to add tests for that.